Import/export umbilicus files

JamesDarby345 opened the pull request:
Added ability to import/export umbilicus files from .obj or .txt files via the GUI in Khartes

JamesDarby345 committed:
import/export umbilicus files + small fixes

KhartesViewer commented:
Thanks for this!  A couple of requests, just to keep the code a bit more tidy.
1) Organize exportUmbilicusFragment as a class, along the lines of InfillDialog (the goal being to try and keep the code out of the MainWindow class, which is already over-packed)
2) Move all detailed I/O operations (writing individual points to a file, reading individual points from a file) out of main_window.py into umbilicus_fragment.py.
Hope that's ok!

JamesDarby345 committed:
move umbilicus code out of main_window

JamesDarby345 commented:
Moved the umbilicus code out of the main window to the umbilicus fragment class + testing import/export quickly, looks good now. Also added random color instead of black when making a new umbilicus

JamesDarby345 committed:
importer to umbilicus file

KhartesViewer commented:
Looks good.  Thanks!
